### ISSUE TITLE
Update configuration-layer/create-layer for changes to layer structure

### ISSUE THREAD

**Issue by chrisbarrett** [syl20bnr/spacemacs#4488]
Issue to track upcoming PR.

`configuration-layer/create-layer` still uses an older style template and includes a redundant extensions.el file. Update the templates create layer command.

**Comment by chrisbarrett** [comment]
A question: as a user, I often create my own packages, but the template's copyright assignment and authorship is always to syl and contributors. Should this be templated to use `user-full-name` instead?

**Comment by chrisbarrett** [comment]
The package list variables are currently assigned using `setq`. I'm going to strongly recommend we use `defconst` to declare the layer packages list variables. :)

`defconst` is semantically similar to `setq` here, but it includes source file metadata in the variable declaration, so it will be possible to navigate to the declaration from help buffers.

**Comment by robbyoconnor** [comment]
:+1:

**Comment by TheBB** [comment]
Great, indeed it looks like `defconst` avoids the problem we had with `defvar`.

**Comment by syl20bnr** [comment]
> but the template's copyright assignment and authorship is always to syl and contributors.

TBH I don't like the current header regarding PRs, I will change it for something lighter when updating to 2016.

```
;; Copyright (c) 2012-2016 Sylvain Benner & Contributors
;;
;; Author: <insert your name here>
```

Should be less pompous :-)
In the reviews we should pay attention to the author line and tell to the contributors to put their name if they want/forget.

**Comment by syl20bnr** [comment]
Fixed in #4491
